src/profile-runner.js

```diff
@@ -42,10 +43,11 @@ class ProfileRunner extends EventEmitter {
   */
   runProfile(profile) {
     this.emit(events.PROFILE_START, profile);
+    const data = testdata(profile.testDataType, this.magnitude);
     const result = {
       profile,
       testResults: profile.functions.map((func) =>
-        this.runFunction(profile, func))
+        this.runFunction(profile, func, data))
```

Comment:
What if the test mutates the data? Please grab a new data set for each test.

Reply:
Will do. But I'll copy the data from line 46 in order to provide the same set of testdata for each iteration.

Reply:
Nevermind. We'll grab a new set of test data each time.